test/test_torch.py

```diff
@@ -14965,6 +14965,19 @@ def test_cat(self, device, dtype):
         self.assertRaises(RuntimeError, lambda: torch.cat([]))
         self.assertRaisesRegex(TypeError, 'got None', lambda: torch.cat([x, None]))
 
+    @onlyCPU
+    def test_cat_different_dtypes(self, device):
+        x = torch.tensor([1, 2, 3], device=device, dtype=torch.int8)
+        y = torch.tensor([4, 5, 6], device=device, dtype=torch.int32)
+        expected_out = torch.tensor([1, 2, 3, 4, 5, 6], device=device, dtype=torch.int32)
+        out = torch.cat([x, y])
+        self.assertEqual(out, expected_out) 
+        z = torch.tensor([7, 8, 9], device=device, dtype=torch.int16)
+        expected_out = torch.tensor([1, 2, 3, 4, 5, 6, 7, 8, 9],
+                                    device=device, dtype=torch.int32)
+        out = torch.cat([x, y, z])
+        self.assertEqual(out, expected_out) 
```

Comment:
there's an exact_dtype option on assertEqual we could use for validating the expected dtype as well